temporalio/worker/_workflow_instance.py

```diff
@@ -606,10 +611,16 @@ async def run_update() -> None:
             finally:
                 self._in_progress_updates.pop(job.id, None)
 
-        self.create_task(
+        task = self.create_task(
             run_update(),
             name=f"update: {job.name}",
         )
+        self._in_progress_updates[job.id] = HandlerExecution(
+            job.name,
+            task,
+            temporalio.workflow.HandlerUnfinishedPolicy.WARN_AND_ABANDON,
+            job.id,
+        )
```

Comment:
We are now storing `Task` in the `HandlerExecution` dataclass, in order to get hold of unfinished update handler tasks for cancellation.